pkg/network/ebpf/c/prebuilt/offset-guess.c

```diff
@@ -10,6 +10,38 @@
 #include <uapi/linux/tcp.h>
 #include <uapi/linux/ip.h>
 
+#define sizeof_member(T, m) sizeof(((T*)0)->m)
+
+static const __u8 SIZEOF_SADDR = sizeof_member(tracer_status_t, saddr);
+static const __u8 SIZEOF_DADDR = sizeof_member(tracer_status_t, daddr);
+static const __u8 SIZEOF_FAMILY = sizeof_member(tracer_status_t, family);
+static const __u8 SIZEOF_SPORT = sizeof_member(tracer_status_t, sport);
+static const __u8 SIZEOF_DPORT = sizeof_member(tracer_status_t, dport);
+static const __u8 SIZEOF_NETNS = sizeof((struct net*)0);
+static const __u8 SIZEOF_NETNS_INO = sizeof_member(tracer_status_t, netns);
+static const __u8 SIZEOF_RTT = sizeof_member(tracer_status_t, rtt);
+static const __u8 SIZEOF_RTT_VAR = sizeof_member(tracer_status_t, rtt_var);
+static const __u8 SIZEOF_DADDR_IPV6 = sizeof_member(tracer_status_t, daddr_ipv6) / 4;
+static const __u8 SIZEOF_SADDR_FL4 = sizeof_member(tracer_status_t, saddr_fl4);
+static const __u8 SIZEOF_DADDR_FL4 = sizeof_member(tracer_status_t, daddr_fl4);
+static const __u8 SIZEOF_SPORT_FL4 = sizeof_member(tracer_status_t, sport_fl4);
+static const __u8 SIZEOF_DPORT_FL4 = sizeof_member(tracer_status_t, dport_fl4);
+static const __u8 SIZEOF_SADDR_FL6 = sizeof_member(tracer_status_t, saddr_fl6) / 4;
+static const __u8 SIZEOF_DADDR_FL6 = sizeof_member(tracer_status_t, daddr_fl6) / 4;
+static const __u8 SIZEOF_SPORT_FL6 = sizeof_member(tracer_status_t, sport_fl6);
+static const __u8 SIZEOF_DPORT_FL6 = sizeof_member(tracer_status_t, dport_fl6);
+static const __u8 SIZEOF_SOCKET_SK = sizeof(uintptr_t);
```

Comment:
After reading up on `uintptr_t`, I think you want `sizeof(void*)` for pointers.

Reply:
I always though `uintptr_t` was the "size of a pointer". I guess practically it is, but there is wiggle room in the standard (as always).

Reply:
C standards 🤝 wiggle room